docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_configuredassetdataconnector.rst

```diff
@@ -31,23 +30,95 @@ All of the examples below assume you’re testing configuration using something
 
     context.test_yaml_config("""
-    class_name: Datasource
-
-    execution_engine:
+    my_data_source:
+      class_name: Datasource
+      execution_engine:
         class_name: PandasExecutionEngine
-
-    data_connectors:
+      data_connectors:
         my_filesystem_data_connector:
-            {data_connector configuration goes here}
+          {data_connector configuration goes here}
     """)
 
 
 If you’re not familiar with the ``test_yaml_config`` method, please check out: :ref:`How to configure DataContext components using test_yaml_config. <how_to_guides_how_to_configure_datacontext_components_using_test_yaml_config>`
 
+Choose a DataConnector
+----------------------
+
+ConfiguredAssetDataConnectors like  ``ConfiguredAssetFilesystemDataConnector`` and ``ConfiguredAssetS3DataConnector`` require DataAssets to be
+explicitly named. Each DataAsset can have their own regex ``pattern`` and ``group_names``, and if configured, will override any
+``pattern`` or ``group_names`` under ``default_regex``.
+
+Imagine you have the following files in ``my_directory/``:
+
+.. code-block:: bash
+
+    my_directory/alpha-1.csv
+    my_directory/alpha-2.csv
+    my_directory/alpha-3.csv
+
+
+We could create a DataAsset ``alpha`` that contains 3 data_references (``alpha-1.csv``, ``alpha-2.csv``, and ``alpha-3.csv``).
+In that case, the configuration would look like the following:
+
+.. code-block:: yaml
+
+    my_data_source:
+      class_name: Datasource
+      execution_engine:
+        class_name: PandasExecutionEngine
+      data_connectors:
+        my_filesystem_data_connector:
+          class_name: ConfiguredAssetFilesystemDataConnector
+          base_directory: my_directory/
+          default_regex:
+          assets:
+            alpha:
+              pattern: alpha-(.*)\.csv
+              group_names:
+                - index
+
+Notice that we have specified a pattern that captures the number after ``alpha-`` in the filename and assigns it to the ``group_name`` ``index``.
+
+The configuration would also work with a regex capturing entire filename (ie ``pattern: (.*)\\.csv``).  However, capturing the index on its own allows for ``partition_identifiers`` to be used to retrieve a specific partition of the DataAsset.
+
+Later on we could retrieve the data in ``alpha-2.csv`` of ``alpha`` as its own batch using ``context.get_batch()`` by specifying ``{"index": "2"}`` as the ``partition_identifier``.
+
+.. code-block:: python
+
+    my_batch = context.get_batch(
+        datasource_name="my_data_source",
+        data_connector_name="my_filesystem_data_connector",
+        data_asset_name="alpha",
+        partition_identifiers={"index": "2"}
+        )
+
+
+This ability to access specific partitions using ``partition_identifiers`` is very useful when validating DataAssets that span multiple files.
+For more information on ``batches`` and ``partition_identifiers``, please refer to the :ref:`Core Concepts document. <reference__core_concepts>`
```

Comment:
Not sure if this belongs here.. should partition_identifiers, splitting, sampling etc be in their own how-to?